core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.html

```diff
@@ -0,0 +1,601 @@
+                  </div>
+                  <div>
+                    {{row.topicName}}
+                  </div>
+                </div>
+                <div class="mobile-stats-row">
+                  <div class="mobile-stats-name">
+                    <b>{{COLUMNS[type][2].name}}</b>
+                  </div>
+                  <div>
+                    {{row.reviewedQuestions}}
+                  </div>
+                </div>
+                <div class="mobile-stats-row">
+                  <div class="mobile-stats-name">
+                    <b>{{COLUMNS[type][3].name}}</b>
+                  </div>
+                  <div>
+                    {{row.acceptedQuestions}}
+                  </div>
+                </div>
+              </div>
+            </div>
+            <div class="oppia-pagination-container">
+              <button type="button"
+                      class="oppia-pagination-button"
+                      (click)="previousPage(block)"
+                      [disabled]="block.currentPage === 1">
+                <span class="material-icons">navigate_before</span>Previous
+              </button>
+              <button type="button"
+                      class="oppia-pagination-button"
+                      (click)="nextPage(block)"
+                      [disabled]="block.currentPage === block.totalPages">
+                Next<span class="material-icons">navigate_next</span>
+              </button>
+            </div>
+        </mat-card>
+        <mat-card *ngIf="statsData[type].length === 0" layout="row" class="list-view-item oppia-contributions-required-message-container">
+          <div class="oppia-contributions-required-message">
+            Start contributing to see your stats!
+          </div>
+        </mat-card>
+    </div>
+</div>
+
+<style>
+  .oppia-contributions-required-message-container {
+    height: 300px;
+  }
+  .oppia-contributions-required-message {
+    color: #4a4a4a;
+    font-size: 28px;
+    text-align: center;
+    vertical-align: middle;
+    height: 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+  .oppia-pagination-container {
+    display: flex;
+    justify-content: space-between;
+    margin: 10px 0;
+    width: 100%;
+  }
+  .oppia-pagination-button {
+    background-color: transparent;
+    border: none;
+    box-sizing: border-box;
+    color: #000;
+    font-size: 18px;
+    font-weight: bold;
+    min-height: 40px;
+    min-width: 100px;
+    text-align: center;
+  }
+  .oppia-pagination-button:hover {
+    border: 2px solid #000;
+    border-radius: 5px;
+  }
+  .oppia-pagination-button[disabled] {
+    color: #000;
+    opacity: 0.239;
+  }
+  .oppia-contributions-dropdown-arrow {
+    color: #009688;
+  }
+  .stats-contribution-type-mobile-container {
+    display: none;
+    background-color: #ffffff;
+    font-size: 20px;
+    margin-bottom: 3%;
+    padding: 0 20px;
+    width: 100%;
+  }
+  .mobile-stats-block {
```

Comment:
Can you do an experiment to see whether, if you use this class name in another component that's still on the CD page, whether the styles "leak out"? You can put "color: red" here or similar so that it's more obvious. Please submit an image as proof of the results when replying to this comment, so that we can verify that you've done the right thing.

If the styles leak out, then you'll need to prefix all these CSS selectors by the name of your component or something that scopes them to being within your component only. Ditto for other HTML files.

Reply:
Er, I'm not sure you understood my comment, let me try again.

I'm saying, try using .mobile-stats-block in another component that's not this one, but that is loaded on the CD page. See if the other component takes the same style.

The thing you're highlighting here seems like a library component and not something we declare. I want to make sure that custom components we declare don't "bleed" into other components -- if they do then we should safeguard against that. Makes sense?

Reply:
Yup, that's great. Thank you for checking!